src/components/Dropdowns/ColorRampDropdown.tsx

```diff
@@ -58,51 +149,57 @@ function arrayDeepEquals<T>(arr1: T[], arr2: T[]): boolean {
   return true;
 }
 
-/** A dropdown selector for color ramp gradients. */
-const ColorRampSelector: React.FC<ColorRampSelectorProps> = (propsInput): ReactElement => {
-  const props = { ...defaultProps, ...propsInput } as Required<ColorRampSelectorProps>;
+export default function ColorRampSelection(inputProps: ColorRampSelectionProps): ReactElement {
+  const props = { ...defaultProps, ...inputProps } as Required<ColorRampSelectionProps>;
   const theme = useContext(AppThemeContext);
 
   const { colorRampsToDisplay, categoricalPalettesToDisplay } = props;
 
-  // TODO: Consider refactoring this into a shared hook if this behavior is repeated again.
-  // Override the open/close behavior for the dropdown so it's compatible with keyboard navigation.
-  const [forceOpen, setForceOpen] = useState(false);
-  const componentContainerRef = React.useRef<HTMLDivElement>(null);
+  ///////// Generate dropdown contents
 
-  // If open, close the dropdown when focus is lost.
-  useEffect(() => {
-    if (!forceOpen) {
-      return;
-    }
-    const doesContainTarget = (target: EventTarget | null): boolean => {
-      return (
-        (target instanceof Element &&
-          componentContainerRef.current &&
-          componentContainerRef.current.contains(target)) ||
-        false
-      );
-    };
-    // Handle focus loss for tab navigation
-    const handleFocusLoss = (event: FocusEvent): void => {
-      if (!doesContainTarget(event.relatedTarget)) {
-        setForceOpen(false);
-      }
-    };
-
-    componentContainerRef.current?.addEventListener("focusout", handleFocusLoss);
-    return () => {
-      componentContainerRef.current?.removeEventListener("focusout", handleFocusLoss);
-    };
-  }, [forceOpen]);
+  const rampItems: SelectItem[] = useMemo(
+    () =>
+      colorRampsToDisplay.map((key) => {
+        const rampData = props.knownColorRamps.get(key);
+        if (!rampData) {
+          throw new Error(`Invalid color ramp key '${key}'`);
+        }
+        return {
+          value: key,
+          label: rampData.name,
+          image: rampData.colorRamp.createGradientCanvas(120, theme.controls.height).toDataURL(),
+          tooltip: rampData.name,
+        };
+      }),
```

Comment:
optional: consider giving this arrow function `{}`s to make the scope a bit clearer (I assume the `map` call was auto-formatted onto the next line?)